File: chatGPT.php
<?php

// $apiKey = $_SERVER['OPENAI_API_KEY'];
// $url = 'https://api.openai.com/v1/chat/completions';
$apiKey = $_SERVER['DEEPSEEK_API_KEY'];
$url = 'https://api.deepseek.com/chat/completions';
// $url = 'https://api.deepseek.com';

$headers = array(
    'Content-Type: application/json',
    'Authorization: Bearer ' . $apiKey
);

$messages = json_decode($_POST['chatBuffer'], true);

$model = json_decode($_POST['model'], true);
$temperature = json_decode($_POST['temperature'], true);
$style = json_decode($_POST['style'], true);
$details = json_decode($_POST['details'], true);

/*
echo "before";
print_r($apiKey);
print_r($messages);
exit;
*/

  $data = array(
      'model' => "deepseek-chat", // $model, "deepseek-chat", "deepseek-reasoner",
      'messages' => $messages,
      'max_tokens' => 1000, // 3000
      'temperature' => 1.3, // $temperature,
      // 'seed' => 12321 // any integer
      // 'top_p' => 0.5
  );
// }

$options = array(
    'http' => array(
        'header' => implode("\r\n", $headers),
        'method' => 'POST',
        'content' => json_encode($data)
    )
);

$context = stream_context_create($options);
$response = file_get_contents($url, false, $context);

// Process the response
if ($response === false) {
    echo 'Error reading response from open a i   A P I.';
} else {
    $responseData = json_decode($response, true);
    // Access the assistant's reply
    $assistantReply = end($responseData['choices'])['message']['content'];
    echo $assistantReply;
}
?>

File: chatGPT.py
#!/usr/bin/env python3

import os
import requests
import json
import cgi

API_KEY = os.getenv("OPENAI_API_KEY")
URL = "https://api.openai.com/v1/responses"

headers = {
    "Authorization": f"Bearer {API_KEY}",
    "Content-Type": "application/json"
}

form = cgi.FieldStorage()

messages = json.loads(form.getvalue("chatBuffer"))
model = "gpt-4o"   # ou "gpt-5"
temperature = 0.7

# Conversion messages -> texte
input_text = ""
for m in messages:
    input_text += f"{m['role'].upper()}: {m['content']}\n"

print(input_text)


payload = {
    "model": model,
    "input": input_text,
    "max_output_tokens": 1000,
    "temperature": temperature
}

response = requests.post(URL, headers=headers, json=payload)
response.raise_for_status()

data = response.json()

# Extraction texte
output = ""
for item in data["output"]:
    if item["type"] == "message":
        for c in item["content"]:
            if c["type"] == "output_text":
                output += c["text"]

print(output)
